src/main/java/uk/gov/justice/digital/hmpps/cmd/api/service/NotificationService.kt

```diff
@@ -234,8 +253,11 @@ class NotificationService(
 
     companion object {
 
-        data class Key(val shiftDate: LocalDate, val shiftType: String, val quantumId: String)
-        fun ShiftNotification.compoundKey() = Key(this.shiftDate, this.shiftType.toUpperCase(), this.quantumId.toUpperCase())
+        data class Key(val quantumId: String, val shiftDate: LocalDate, val shiftType: String, val taskStart: Long? )
+        fun ShiftNotification.compoundKey() = Key(this.quantumId.toUpperCase(), this.shiftDate, this.shiftType.toUpperCase(), this.taskStart)
+
+        data class ShiftTaskSaveKey(val quantumId: String, val shiftDate: LocalDate, val shiftModified: LocalDateTime, var taskStart: Long)
+        fun ShiftNotificationDto.compoundKey() = ShiftTaskSaveKey(this.quantumId.toUpperCase(), this.actionDate, this.shiftModified, this.taskStart!!)
```

Comment:
I have added task start here, because we were matching shift notifications and shift task notifications with the same modified time. (+ written a test)